src/main/java/bio/terra/app/configuration/scheduled/ShedlockConfiguration.java

```diff
@@ -0,0 +1,48 @@
+package bio.terra.app.configuration.scheduled;
+
+import bio.terra.app.configuration.DataRepoJdbcConfiguration;
+import net.javacrumbs.shedlock.core.LockProvider;
+import net.javacrumbs.shedlock.provider.jdbctemplate.JdbcTemplateLockProvider;
+import net.javacrumbs.shedlock.spring.annotation.EnableSchedulerLock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+/**
+ * ShedLock allows us to safely use Spring's @Scheduled annotation for scheduling background jobs
+ * within a multi-instance environment.
+ *
+ * <p>Spring, by default, cannot handle scheduler synchronization over multiple instances. It
+ * executes the jobs simultaneously on every node instead.
+ *
+ * <p>ShedLock makes sure that your scheduled tasks are executed at most once at the same time. If a
+ * task is being executed on one node, it acquires a lock which prevents execution of the same task
+ * from another node or thread. In the presence of an active lock, execution on other nodes is
+ * skipped.
+ *
+ * <p>More information: https://github.com/lukas-krecan/ShedLock#jdbctemplate
+ */
+@Configuration
+@EnableScheduling
+@EnableSchedulerLock(defaultLockAtMostFor = "1h")
```

Comment:
Is there a way to tie the fact that this is enabled to a config property (just wondering if we want to avoid scheduling in non-prod environments)

Reply:
I think we could add in a `@ConditionalOnProperty` annotation here:

https://github.com/DataBiosphere/jade-data-repo/blob/243b1c18f1e593fcef28ba53a85ff0bc2f5b7f48/src/main/java/bio/terra/app/controller/SearchApiController.java#L42

But I'll also flag this section within the new README, which explains how individual `@Scheduled` tasks can be disabled on default.  Would you still prefer that we also enable global scheduling by environment?

> The following example disables our job by default using Spring's cron disable expression: `-`.
>
> ```java
>@Scheduled(cron = "${jobs.cronSchedule:-}")
>public void cleanTempDirectory() {
>  // do work here
>}
>```
>
> If we wanted to enable our job, we would need to provide a valid cron expression for
`jobs.cronSchedule` via environment variable, property file, etc.

Reply:
oh derp.  you're totally right.  thanks!